### PR TITLE
Jax devicearray diagonal

### LINKED ISSUE #15371: diagonal

### PR DESCRIPTION
**Pull request by VictorOdede**
Close #15371

### CODE REVIEW

**Comment by VictorOdede** [comment]
@hirwa-nshuti I've fixed the merge conflicts for this branch.

**Review by hirwa-nshuti — APPROVED** [review]
Looks good to me 😊
Going to merge🙂

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Would you resolve the merge conflicts

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 1]
```diff
@@ -147,6 +147,42 @@ def test_jax_devicearray_copy(
...
+handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="jax.numpy.array",
+    method_name="diagonal",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        min_num_dims=2,
+    ),
+)
+
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:159]:
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 2]
```diff
@@ -147,6 +147,42 @@ def test_jax_devicearray_copy(
...
+
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:160]:
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 3]
```diff
@@ -147,6 +147,42 @@ def test_jax_devicearray_copy(
...
+def test_jax_devicearray_diagonal(
+    dtype_and_x,
+    on_device,
+    frontend,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+):
+    input_dtype, x = dtype_and_x
+
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:170]:
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 4]
```diff
@@ -147,6 +147,42 @@ def test_jax_devicearray_copy(
     )
 
 
+handle_frontend_method(
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:150]:
```suggestion
@handle_frontend_method(
```

**Pull request merged**